src/modules/generateRandomNumber.js

```diff
@@ -7,7 +7,21 @@
  * @return {number} A random 4-digit number
  */
 function generateRandomNumber() {
-  /* Write your code here */
+  let resultStr = '';
+
+  while (resultStr.length < 4) {
+    const randomDigitStr = String(Math.floor(Math.random() * 10));
+
+    if (resultStr.length === 0) {
+      if (randomDigitStr !== '0') {
+        resultStr += randomDigitStr;
+      }
+    } else if (!resultStr.includes(randomDigitStr)) {
+      resultStr += randomDigitStr;
+    }
+  }
+
+  return +resultStr;
```

Comment:
The function returns a number by converting the string `resultStr` to a number using the unary plus operator. This is correct as per the function's documentation, which specifies that it should return a number. Ensure that this conversion is intentional and aligns with the rest of your application's logic.